project_euler/problem_135/sol1.py

```diff
@@ -0,0 +1,56 @@
+"""
+Project Euler Problem 135: https://projecteuler.net/problem=135
+
+Given the positive integers, x, y, and z,
+are consecutive terms of an arithmetic progression,
+the least value of the positive integer, n,
+for which the equation,
+x2 − y2 − z2 = n, has exactly two solutions is n = 27:
+
+342 − 272 − 202 = 122 − 92 − 62 = 27
+
+It turns out that n = 1155 is the least value
+which has exactly ten solutions.
+
+How many values of n less than one million
+have exactly ten distinct solutions?
+
+
+Taking x,y,z of the form a+d,a,a-d respectively,
+the given equation reduces to a*(4d-a)=n.
+Calculating no of solutions for every n till 1 million by fixing a
+,and n must be multiple of a.
+Total no of steps=n*(1/1+1/2+1/3+1/4..+1/n)
+,so roughly O(nlogn) time complexity.
+
+"""
+
+
+def solution(limit: int = 1000000) -> int:
+    """
+    returns the values of n less than or equal to the limit
+    have exactly ten distinct solutions.
+    >>> solution(100)
+    0
+    >>> solution(10000)
+    45
+    >>> solution(50050)
+    292
+    """
+    limit = limit + 1
+    freq = [0] * limit
```

Comment:
Descriptive variable names than `freq`, `a`, `n`, `d`

Reply:
can change freq to frequency but a and d as mentioned in my solution approach are standard ways of taking the first term and common difference in an A.P and n is mentioned in the question.

Reply:
Yes for `freq`.
I don't know what does A.P. means but as we're coding in Python and also there are some standards we follow in this repository to improve the readability and maintainability of code, we require descriptive variable names.

Why? This code might be looked by thousands of other people and they might wonder what does `a` or `d` even mean? They would have to do some googling or what not. So, maybe it is better to provide a name which is self explanatory :)